api/v1/resourceslice.go

```diff
@@ -44,6 +44,25 @@ type ResourceState struct {
 	Deleted    bool         `json:"deleted,omitempty"`
 }
 
+func (r *ResourceState) Equal(rr *ResourceState) bool {
```

Comment:
Does DeepEqual not work here? Or do we care about these semantics for some concrete reason?

Reply:
It would work, I'm just concerned about perf since this is called for every resource every time the slice changes